### PR TITLE
bug fix : events can be removed by event admins or org admins only

### LINKED ISSUE #261: [Security Bug] User has access to delete others events (API)
**Describe the bug**
Normal user can delete events created by other user/admin by calling API

**To Reproduce**

1. Switch to Events Screen
2. Tap on Event (which is created by other user)
3. Click on menu (3 dot) & Tap delete.
Or 
Call Delete Event API from any user

**Expected behavior**
Delete event API call should be Denied if call it not from original Creator/Admin

**Actual behavior**
Normal User has access to delete others events

**Screenshots**

**Additional details**
1) Front-end part will be fixed in PalisadoesFoundation/talawa#832
2) Backend API part need to address in this issue. So, non-creators will not able to delete event by using services like Postman.
(This might get fixed by doing some updates in `talawa-api/resolvers/event_mutations/removeEvent.js`) 

Server Side Logic :
119254871-68c3f400-bbd6-11eb-9b38-066932057ea6

### PR DESCRIPTION
**Pull request by akshatgarg12**
**What kind of change does this PR introduce?**
fixed #261
bugfix : events can now be only removed by the the event admins and organisation admins

**Did you add tests for your changes?**
Yes, tested on graphiql interface

**If relevant, did you update the documentation?**

**Summary**

Solves a security bug ,prevents user from deleting events they didn't create.

**Does this PR introduce a breaking change?**
nope

**Other information**
I had to use the .toString() method to compare the objectIds as some were present in type string and some in ObjectId

### CODE REVIEW

File: resolvers/event_mutations/removeEvent.js [hunk 1]
```diff
@@ -11,8 +11,21 @@ const removeEvent = async (parent, args, context) => {
   const event = await Event.findOne({ _id: args.id });
   if (!event) throw new Error('Event not found');
 
-  if (!(event.creator !== context.userId)) {
-    throw new Error("User cannot delete event they didn't create");
+  const isUserOrganisationAdmin =
```
Comment (sumitra19jha) [inline resolvers/event_mutations/removeEvent.js:14]: Use `includes()`

File: resolvers/event_mutations/removeEvent.js [hunk 2]
```diff
@@ -11,8 +11,21 @@ const removeEvent = async (parent, args, context) => {
...
+    user.adminFor.filter(
+      (organisationId) =>
+        organisationId.toString() === event.organization.toString()
+    ).length > 0;
+
+  const isUserEventAdmin =
```
Comment (sumitra19jha) [inline resolvers/event_mutations/removeEvent.js:20]: Use `includes()`

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please fix the conflict in the file: `resolvers/event_mutations/removeEvent.js`

**Comment by akshatgarg12** [comment]
@palisadoes @sumitra19jha done!

File: resolvers/event_mutations/removeEvent.js [hunk 3]
```diff
@@ -11,8 +11,16 @@ const removeEvent = async (parent, args, context) => {
   const event = await Event.findOne({ _id: args.id });
   if (!event) throw new Error('Event not found');
```
Comment (impiyush83) [inline resolvers/event_mutations/removeEvent.js:12]: Use standardized error classes

File: resolvers/event_mutations/removeEvent.js [hunk 4]
```diff
@@ -11,8 +11,16 @@ const removeEvent = async (parent, args, context) => {
...
+  const isUserOrganisationAdmin =
+    user.adminFor.includes(event.organization.toString());
+
+  const isUserEventAdmin =
+    event.admins.includes(context.userId.toString());
+    
+  const userCanDeleteThisEvent = isUserOrganisationAdmin || isUserEventAdmin;
+
+  if (!userCanDeleteThisEvent) {
+    throw new Error("Non-Admin User cannot delete event they didn't create");
```
Comment (impiyush83) [inline resolvers/event_mutations/removeEvent.js:23]: Use standardized error classes

**Review by impiyush83 — CHANGES_REQUESTED** [review]
@akshatgarg12   Update your master with latest master.

File: resolvers/event_mutations/removeEvent.js (hunk 3 above)
Comment (akshatgarg12) [reply resolvers/event_mutations/removeEvent.js]: @impiyush83  yes done

**Comment by palisadoes** [comment]
@DMills27
@xsh2047
@Laurell876 

Please review and merge

**Review by impiyush83 — APPROVED** [review]

**Review by sumitra19jha — APPROVED** [review]

**Comment by ranilwallace** [comment]
Please resolve all conversations so we can merge.

**Review by ranilwallace — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@akshatgarg12  you need to resolve the comments from @sumitra19jha before this can be merged.

**Review by palisadoes — APPROVED** [review]

**Comment by akshatgarg12** [comment]
> @akshatgarg12 [you need to resolve the comments](https://github.com/PalisadoesFoundation/talawa-api/pull/271/files/6ce45ab0bb9a831a40f9f0eb60b24b00dc2dc7ac#r641988882) from @sumitra19jha before this can be merged.

done.

**Pull request merged**